### PR TITLE
Mond/refactor internet LM-206

### PR DESCRIPTION
**Pull request by Chawak**
## Why this PR
Why we need this PR? 
This PR is for this issue 
[https://linear.app/openthaigpt/issue/LM-206/refactor-common-crawl-dataset-pipeline-to-use-the-latest-metadatajson
](https://linear.app/openthaigpt/issue/LM-206/refactor-common-crawl-dataset-pipeline-to-use-the-latest-metadatajson
)
## Changes
- Write some changes here

## Related Issues
Close #

## Checklist
- [ ] PR should be in the [Naming convention](../../docs/PR_NAMING.md)
- [ ] Assign yourself in to Assigneees
- [ ] Tag related issues
- [ ] Constants name should be ALL_CAPITAL, function name should be snake_case, and class name should be CamelCase
- [ ] complex function/algorithm should have [Docstring](https://peps.python.org/pep-0257/)
- [ ] 1 PR should not have more than 200 lines changes (Exception for test files). If more than that please open multiple PRs
- [ ] At least PR reviewer must come from the task's team (model, eval, data)

### CODE REVIEW

**Review by new5558 — APPROVED** [review]

File: src/data/pyproject.toml [hunk 1]
```diff
@@ -24,6 +24,7 @@ dependencies = [
     "datasketch<2.0.0",
     "nlpo3 >=1.3.0, <2",
     "beautifulsoup4==4.12.2, < 5",
-    "scikit-learn==1.2.2, < 2"
+    "scikit-learn==1.2.2, < 2",
+    "gitpython"
```
Comment (new5558) [inline src/data/pyproject.toml:28]: Version?

File: src/data/scripts/internet/main.py [hunk 2]
```diff
@@ -286,8 +272,9 @@ def read_jsonl_zst_files(dir_path):
                 "batch_size": batch_size,
```
Comment (new5558) [inline src/data/scripts/internet/main.py:272]: Isnt adding config_dict["processing_parameters"] without specifying do_perplexity etc. better?

File: src/data/scripts/internet/main.py [hunk 3]
```diff
@@ -273,11 +258,12 @@ def read_jsonl_zst_files(dir_path):
         print("Finish processing")
 
         info = {"source": source, "current_version": version}
-        json.dump(info, open(f"{output_dir}/info.json", "w"))
+        create_info_file(info, output_dir)
 
         metadata = {
             "dataset_name": source,
             "data_version": version,
```
Comment (new5558) [inline src/data/scripts/internet/main.py:265]:
```suggestion
            "dataset_version": version,
```

File: src/data/scripts/internet/main.py [hunk 4]
```diff
@@ -273,11 +258,12 @@ def read_jsonl_zst_files(dir_path):
         print("Finish processing")
 
         info = {"source": source, "current_version": version}
-        json.dump(info, open(f"{output_dir}/info.json", "w"))
+        create_info_file(info, output_dir)
 
         metadata = {
```
Comment (new5558) [inline src/data/scripts/internet/main.py:263]: Can we move the logic to create metadata dict into metadata.py to improve consistency across entire data pipeline

File: src/data/scripts/internet/main.py [hunk 5]
```diff
@@ -12,57 +12,40 @@
...
-    input_based_path = cfg.input_dataset.path
-
-    info = json.load(open(f"{input_based_path}/info.json", "r"))
-    input_version = info["current_version"]
-    source = info["source"]
-
-    input_path = f"{input_based_path}/{input_version}/data"
-
-    print(f"Processing {source} dataset")
-    if "version" in cfg.output:
```
Comment (new5558) [inline src/data/scripts/internet/main.py:53]: Why we remove version bump logic?

File: src/data/scripts/internet/main.py [hunk 6]
```diff
@@ -12,57 +12,40 @@
...
-    do_perplexity = cfg.processing_parameters.do_perplexity
-    batch_size = cfg.processing_parameters.batch_size
-    sampled_back_ratio = cfg.processing_parameters.sampled_back_ratio
-
-    output_dir = cfg.output.path
-    scratch_location = cfg.output.scratch_path if "scratch_path" in cfg.output else None
-
-    input_based_path = cfg.input_dataset.path
-
-    info = json.load(open(f"{input_based_path}/info.json", "r"))
```
Comment (new5558) [inline src/data/scripts/internet/main.py:46]: Why we remove logic to get info from input’s info.json

Comment (Chawak) [reply src/data/scripts/internet/main.py:46]: Just move it to `core/processing_config.py` krub

File: src/data/scripts/internet/main.py (hunk 5 above)
Comment (Chawak) [reply src/data/scripts/internet/main.py:53]: Same as info krub

**Review by new5558 — APPROVED** [review]

**Review by ArthurMinovsky — APPROVED** [review]

**Pull request merged**